src/shell/shell.cpp

```diff
@@ -54,7 +54,7 @@ std::unique_ptr<Program> SHELL_ProgramCreate() {
 	return ProgramCreate<DOS_Shell>();
 }
 
-DOS_Shell::DOS_Shell()
+DOS_Shell::DOS_Shell() : history(std::make_shared<ShellHistory>())
```

Comment:
I'm not sure what the purpose of this is.  The `ShellHistory` object gets created a few lines down from this in the constructor.  With this change, you're creating a redundant object that gets discarded.

We want exactly one `ShellHistory` object so the code is using a `static weak_ptr` to achieve this.  If the `weak_ptr` is empty, that means this shell is either the first shell or all active shells have been destroyed, so we create a new one and store it in the static variable.  Otherwise, we use the existing one stored.

Reply:
(back to original; thanks for catching this @weirddan455.)